source/develop/release-management/features/gluster-self-heal-monitoring.html.md

```diff
@@ -41,21 +41,51 @@ Host Maintenance and Fencing will be enabled/disabled based on the following cri
 | No Unsynced Entries | Quorum available without this Node | Allowed |
 | No Unsynced Entries | Quorum not available without this Node | Not Allowed |
 
+ More info about standard host fencing is available at http://old.ovirt.org/Automatic_Fencing#Automatic_Fencing http://old.ovirt.org/Fence_kdump https://www.youtube.com/watch?v=V1JQtmdleaM
+
 ## Entity Changes
 Following entities will be changed as part of Gluster self heal monitoring.
 
 ### Gluster_volume_bricks
   Following columns will be added to gluster_volume_bricks table.
   unsynced_entries  - integer - No.of unsynced entries in the brick.
   unsynced_entires_history - text - History of unsynced entries in the brick. It will a list of comma separated values.
-  
-### Gluster_volumes_view
-‘heal_status’ column will be be added to gluster_volumes_view. It will be computed based on unsynced_entries in the brick. If there is any unsynced_entry in any of the bricks in the volume then ‘heal_status’ will be ‘Needs_Healing’.
 
+## Change in BLL Commands
+###VdsNotRespondingTreatmentCommand (for Host fencing)
+   VdsNotRespondingTreatmentCommand is the central BLL command which takes care of varios host fencing related activites. This will be changed to consider gluster self-heal information during host fencing. shouldFencingBeSkipped method will be changed and following logic will be added.
+   
+   if (host supports both gluster & virt services)
+      if(all bricks are down)
+        allow host fencing
+      else(some || all bricks are up)
+        if (unsync'd entries  > 0 for any up brick) //Entris needs to be healed from this host
+          skip host fencing
+        else
+          if (quorum is maintained (bricks and nodes for ALL volumes) with this host down)
+            allow host fencing
+          else
+            skip host fencing
+
+###MaintenanceNumberOfVdssCommand (for Host Maintenance)
+  MaintenanceNumberOfVdssCommand is used for moving one or more number of hosts to maintenance. It will be enhanced to consider gluster self-heal and quorum before moving the host to maintenance. Following validations will be added to MaintenanceNumberOfVdssCommand.validate() method.
+  
+     if (host supports both gluster & virt services & not force-maintenance)
```

Comment:
Hmm, this is less harmful than NotRespondingTreatment, but here we have even worse timing issue, because migrating all VMs from the host can take a huge amount of time:

1. If we execute those checks in validate, host may not be use for healing at the moment, but as VMs migration from the hosts may take long time, healing may start during this migration and we have an issue.
2. On the other hand host may be used for healing, but healing may be finished before VMs migration, so we disable move to maintenance operation unnecessarily.

Due to above, wouldn't it be better following approach:
1. Before execution of MaintenanceNumberOfVdssCommand check for healing status and if healing is in progress display warning to user: "Healing process is performed so move to maintenance may not be successfull. Do you really want to continue? Yes/No"
2. If user chooses Yes, than execute MaintenanceNumberOfVdssCommand, but repeat gluster healing check after all VMs migrations are finished and prior to changing host status to Maintenance. And if healing is still runing, just fail the command with error "Host cannot be moved to Maintenance, because gluster healing process is currenlty performed".

What do you think?